charmhub/http.go

```diff
@@ -155,7 +155,7 @@ func (t *apiRequester) Do(req *http.Request) (*http.Response, error) {
 				req.Body = io.NopCloser(bytes.NewReader(body))
 			}
 			var err error
-			resp, err = t.doOnce(req)
+			resp, err = t.doOnce(req) //nolint:bodyclose // resp.Body is closed by the caller.
```

Comment:
Here we set the resp on an object and return that object, requiring the caller to handle the close. This can lead to issues if we're not careful.

Reply:
The callers already do this. It's a bit convoluted but when there is an error, it reads the body to io.Discard and then closes it. Otherwise if there is no error the response gets read by the caller.